Update for signac 2.0 and general cleanup (#183)
glotzerlab/signac-docs: docs/source/projects.rst

Project Initialization
======================

In order to use **signac** to manage a project's data, the project must be initialized as a **signac** project.

> After a project has been initialized in **signac**, all shell and Python scripts executed within or below the project's root directory have access to **signac**'s central facility, the **signac** project interface.

Reviewer: ```suggestion
After a project has been initialized in **signac**, all shell and Python scripts executed within the project path or its subdirectories have access to **signac**'s central facility, the **signac** project interface.
```

The project interface provides simple and consistent access to the project's underlying *data space*. [#f1]_

.. [#f1] You can access a project interface from other locations by explicitly specifying the root directory.

To initialize a project, execute ``$ signac init`` on the command line inside the desired project directory (create a new project directory if needed).
For example, to initialize a **signac** project in a directory called ``my_project``, execute:

.. code-block:: bash

    $ mkdir my_project
    $ cd my_project
    $ signac init

You can alternatively initialize your project within Python with the :py:func:`~signac.init_project` function:

.. code-block:: pycon

    >>> project = signac.init_project()

This will create a ``.signac`` directory with a configuration file.
The directory containing the ``.signac`` subdirectory is the project's root directory.

.. _project-data-space:

The Data Space
==============

> The project data space is stored in the *workspace directory*, a subdirectory within the project's root directory named ``workspace``.

Reviewer: ```suggestion
The project data space is stored in the *workspace directory*, a subdirectory within the project path named ``workspace``.
```

Once a project has been initialized, any data inserted into the data space will be stored within this directory.

> You can access your signac :py:class:`~signac.Project` and the associated data from within your project's root directory or any subdirectory with the :py:func:`~signac.get_project` function:

Reviewer: ```suggestion
You can access your signac :py:class:`~signac.Project` and the associated data from within the project path or any subdirectory with the :py:func:`~signac.get_project` function:
```

.. tip::

    You only need to call the :meth:`Job.init` function the first time that you are accessing a job.
    Furthermore, the :meth:`Job.init` function returns the job itself, so you can abbreviate like this:

    .. code-block:: python

        job = project.open_job({"temperature": 20, "pressure": 1.0}).init()

> The uniqueness of a state point should be familiar to anyone who has used a relational database: the state point parameters constitute the primary key of the data space.

Reviewer 2: Rewrite to be inclusive of people who don't know about relational databases. Something like:
```suggestion
The uniqueness of a state point is like that in a relational database where the state point parameters are the primary key of the data space.
```

Reviewer: Small rewrite to clarify the analogy:
```suggestion
The uniqueness of a state point in a signac data space is analogous to the uniqueness of a primary key in a relational database.
```

---------------------------

In some cases it may desirable to divide or merge a project data space.

> To **move** a job to a different project, use the :py:meth:`~signac.job.Job.move` method:

Author: No, I only edited the ones that were part of code I already had to change in this PR. We can do a batch cleanup later if we'd like.
